resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -249,6 +259,8 @@ private[spark] class Client(
     val capability = Records.newRecord(classOf[Resource])
     capability.setMemory(amMemory + amMemoryOverhead)
     capability.setVirtualCores(amCores)
+    ResourceTypeHelper.setResourceInfoFromResourceTypes(driverResourceTypes, capability)
```

Comment:
Only call if there are resources defined. Otherwise this will print noisy logs for people who are not using the feature.

Reply:
fixed